test/features/public_asset/domain/extensions/string_to_public_asset_extension_test.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/extensions/string_to_public_asset_extension.dart';
+
+void main() {
+  group('string to public asset extension test:', () {
+    test(
+      'should convert string to public assets',
+    () {
+      // arrange
+      const publicAssetIdValue = 'public-asset-1';
+      const htmlString = '<img src="public-asset-1" public-asset-id="$publicAssetIdValue"/>';
```

Comment:
please add more cases, error cases too

Reply:
Please list down every case you want to see in the test.

Reply:
case 1:  More image tag have ` public-asset-id` attribute
case 2: More image tag don't have ` public-asset-id` attribute
case 3: More image tag have both `public-asset-id` attribute and no `public-asset-id`

Reply:
Done